src/log/log.c

```diff
@@ -42,21 +42,21 @@ __wt_log_printf(WT_SESSION_IMPL *session, const char *format, ...)
  *	Given a log record, return whether the checksum matches.
  */
 static bool
-__log_checksum_match(WT_SESSION_IMPL *session, WT_ITEM *buf, uint32_t reclen)
+__log_checksum_match(WT_ITEM *buf, uint32_t reclen)
 {
 	WT_LOG_RECORD *logrec;
-	uint32_t checksum_calculate, checksum_tmp;
+	uint32_t checksum_saved, checksum_tmp;
+	bool checksum_matched;
 
-	WT_UNUSED(session);
-	logrec = (WT_LOG_RECORD *)buf->mem;
-	checksum_tmp = logrec->checksum;
-	logrec->checksum = 0;
-	checksum_calculate = __wt_checksum(logrec, reclen);
+	logrec = buf->mem;
+	checksum_saved = checksum_tmp = logrec->checksum;
 #ifdef WORDS_BIGENDIAN
-	checksum_calculate = __wt_bswap32(checksum_calculate);
+	checksum_tmp = __wt_bswap32(checksum_tmp);
```

Comment:
You've swapped the things being byte-swapped here. It used to be swapping the calculated checksum, now it's byte-swapping the saved checksum. I *think* that results in the same checking, but am pointing it out because I'm not confident enough in my endian knowledge.

Reply:
I <i>think</i> it's the same, too...

@lukech, does our BE/LE testing include log file portability tests?

Reply:
@keithbostic Current BE/LE tests verify data files generated by below test/format command. I don't think log files are generated.
`./t -1 -h "WT_TEST.$i" -c $(pwd)/../../../test/format/CONFIG.endian`

Reply:
fwiw, I think this change is fine.